app/src/main/java/es/wolfi/app/passman/SJCLCrypto.java

```diff
@@ -21,10 +21,159 @@
  */
 package es.wolfi.app.passman;
 
+
+import android.util.Log;
+import android.util.Base64;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import org.json.JSONObject;
+
+import java.security.AlgorithmParameters;
+import java.security.SecureRandom;
+import java.util.Arrays;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
 public class SJCLCrypto {
-    public static native String decryptString(String cryptogram, String key) throws Exception;
+    //public static native String decryptString(String cryptogram, String key) throws Exception;
+
+    private static final String ALGORITHM = "AES/CCM/NoPadding";
+    private static final int TAG_LENGTH_BIT = 64;
+    private static final int SALT_LENGTH_BYTE = 8;
+    private static final int ITER = 1000;
+    private static final int KEY_LENGTH_BIT = 256;
 
-    static {
+    /*static {
         System.loadLibrary("passman-lib");
+    }*/
+
+    public static JSONObject encryptString(String toencrypt, String stringkey)
+    {
+        try {
```

Comment:
This class should perform all the SJCL abstraction, which includes JSON encoding, not only encryption, I comment other issues on the next function.